src/App.vue

```diff
@@ -381,6 +358,10 @@ export default class App extends Mixins(CommonMixin, DateMixin, FilingTemplateMi
     //
 
     if (val) {
+      // store current filing type
+      const filingType = this.$route.matched[0]?.meta.filingType
+      filingType && this.setFilingType(filingType)
```

Comment:
I implemented this more flexible way when I attempted to separate the Company Info page and the Review and Confirm page (with separate routes), but that caused other difficult problems.

Anyway, I left this as I think it's slick.